components/automate-cli/pkg/verifyserver/server/api/v1/softwareversion_test.go

```diff
@@ -0,0 +1,185 @@
+	}
+	app := fiber.New(fconf)
+	handler := v1.NewHandler(log).AddSoftwareVersionService(sv)
+
+	vs := &server.VerifyServer{
+		Port:    server.DEFAULT_PORT,
+		Log:     log,
+		App:     app,
+		Handler: handler,
+	}
+	vs.SetupRoutes()
+	return vs.App
+}
+
+func TestSoftwareVersionAPI(t *testing.T) {
+	tests := []struct {
+		description   string
+		expectedCode  int
+		responseBody  models.SoftwareVersionDetails
+		expectedBody  string
+		expectedError error
+		query         string
+	}{
+		{
+			description:  "200:success software version check route when query is opensearch",
+			expectedCode: 200,
+			responseBody: models.SoftwareVersionDetails{
+				Passed: true,
+				Checks: []models.Checks{
+					{
+						Title:         "openssl availability",
+						Passed:        true,
+						SuccessMsg:    "openssl is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "mkdir availability",
+						Passed:        true,
+						SuccessMsg:    "mkdir is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "Linux Version Check",
+						Passed:        true,
+						SuccessMsg:    "Ubuntu version is 20.04",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedBody:  `{"status":"SUCCESS","result":{"passed":true,"checks":[{"title":"openssl availability","passed":true,"success_msg":"openssl is available","error_msg":"","resolution_msg":""},{"title":"mkdir availability","passed":true,"success_msg":"mkdir is available","error_msg":"","resolution_msg":""},{"title":"Linux Version Check","passed":true,"success_msg":"Ubuntu version is 20.04","error_msg":"","resolution_msg":""}]}}`,
+			expectedError: nil,
+			query:         "opensearch",
+		},
+		{
+
+			description:  "200:success software version check route when query is postgres",
+			expectedCode: 200,
+			responseBody: models.SoftwareVersionDetails{
+				Passed: true,
+				Checks: []models.Checks{
+					{
+						Title:         "stat availability",
+						Passed:        true,
+						SuccessMsg:    "stat is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "mkdir availability",
+						Passed:        true,
+						SuccessMsg:    "mkdir is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "Linux Version Check",
+						Passed:        true,
+						SuccessMsg:    "Ubuntu version is 20.04",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedBody:  `{"status":"SUCCESS","result":{"passed":true,"checks":[{"title":"stat availability","passed":true,"success_msg":"stat is available","error_msg":"","resolution_msg":""},{"title":"mkdir availability","passed":true,"success_msg":"mkdir is available","error_msg":"","resolution_msg":""},{"title":"Linux Version Check","passed":true,"success_msg":"Ubuntu version is 20.04","error_msg":"","resolution_msg":""}]}}`,
```

Comment:
same use  &models.Checks{} and get success/ error message from variable(like u have done for **responseBody**) or create global variable for entire expectedBody